packages/library/src/rpc-request-coalescer.ts

```diff
@@ -29,13 +29,28 @@ export function getRpcTransportWithRequestCoalescing(
         }
         if (coalescedRequestsByDeduplicationKey[deduplicationKey] == null) {
             const abortController = new AbortController();
+            const responsePromise = (async () => {
+                try {
+                    return await transport<TResponse>({
+                        ...config,
+                        signal: abortController.signal,
+                    });
+                } catch (e) {
+                    if (e && typeof e === 'object' && 'name' in e && e.name === 'AbortError') {
+                        // Ignore `AbortError` thrown from the underlying transport behind which all
+                        // requests are coalesced. If it experiences an `AbortError` it is because
+                        // we triggered one when the last subscriber aborted. Letting the underlying
+                        // transport's `AbortError` bubble up from here would cause runtime fatals
+                        // where there should be none.
+                        return;
+                    }
+                    throw e;
+                }
+            })();
```

Comment:
I may be missing something, but wouldn't we want this to just be the `AbortSignal` provided to the transport itself? What's the reason we create a new one here?

Reply:
The coalescing transport works by:

* reference counting all callers with the ‘same request’ (as defined by having the same deduplication key)
* creating an ‘underlying’ request when the number of callers goes from 0 to 1
* aborting the ‘underlying’ request when the number of callers goes from 1 to 0

When you have 2 or more callers and one is aborted, you want:

* the `AbortError` to be thrown at the callsite of the caller that was aborted
* the ‘underlying’ request to keep on chugging

So no; each caller gets a ‘synthetic’ promise for the ‘underlying’ request's response. When that caller aborts, we only want that synthetic to be aborted. We _don't_ want the ‘underlying’ request to be aborted until there are no more callers waiting on it, at which point we should abort it quietly and without impact to the rest of the app.